internal/armada/server/submit.go

```diff
@@ -387,97 +454,46 @@ func (server *SubmitServer) reprioritizeJobs(jobIds []string, newPriority float6
 	if err != nil {
-		return err
+		return fmt.Errorf("[reportReprioritizedJobEvents] error reporting jobs reprioritized: %w", err)
 	}
 
 	return nil
 }
 
-func (server *SubmitServer) checkReprioritizePerms(ctx context.Context, jobs []*api.Job) error {
-	queues := make(map[string]bool)
-	for _, job := range jobs {
-		queues[job.Queue] = true
-	}
-	for queue := range queues {
-		if e, _ := server.checkQueuePermission(ctx, queue, false, permissions.ReprioritizeJobs, permissions.ReprioritizeAnyJobs); e != nil {
-			return e
-		}
-	}
-	return nil
-}
-
-func (server *SubmitServer) checkQueuePermission(
-	ctx context.Context,
-	queueName string,
-	attemptToCreate bool,
-	basicPermission permission.Permission,
-	allQueuesPermission permission.Permission) (e error, ownershipGroups []string) {
-
-	queue, e := server.queueRepository.GetQueue(queueName)
-	if e == repository.ErrQueueNotFound {
-		if attemptToCreate &&
-			server.queueManagementConfig.AutoCreateQueues &&
-			server.permissions.UserHasPermission(ctx, permissions.SubmitAnyJobs) {
-
-			queue = &api.Queue{
-				Name:           queueName,
-				PriorityFactor: server.queueManagementConfig.DefaultPriorityFactor,
-			}
-			e := server.queueRepository.CreateQueue(queue)
-			if e != nil {
-				return status.Errorf(codes.Aborted, e.Error()), []string{}
-			}
-			return nil, []string{}
-		} else {
-			return status.Errorf(codes.NotFound, "Queue %q not found", queueName), []string{}
-		}
-	} else if e != nil {
-		return status.Errorf(codes.Unavailable, "Could not load queue %q: %s", queueName, e.Error()), []string{}
-	}
-
-	permissionToCheck := basicPermission
-	owned, groups := server.permissions.UserOwns(ctx, queue)
-	if !owned {
-		permissionToCheck = allQueuesPermission
-	}
-	if e := checkPermission(server.permissions, ctx, permissionToCheck); e != nil {
-		return e, []string{}
-	}
-	return nil, groups
-}
-
+// createJobs returns a list of objects representing the jobs in a JobSubmitRequest.
+// This function validates the jobs in the request and the pod specs. in each job.
+// If any job or pod in invalid, an error is returned.
 func (server *SubmitServer) createJobs(request *api.JobSubmitRequest, owner string, ownershipGroups []string) ([]*api.Job, error) {
 	jobs := make([]*api.Job, 0, len(request.JobRequestItems))
 
 	if request.JobSetId == "" {
-		return nil, fmt.Errorf("job set is not specified")
+		return nil, fmt.Errorf("[createJobs] job set not specified")
 	}
 
 	if request.Queue == "" {
-		return nil, fmt.Errorf("queue is not specified")
+		return nil, fmt.Errorf("[createJobs] queue not specified")
 	}
 
 	for i, item := range request.JobRequestItems {
 
 		if item.PodSpec != nil && len(item.PodSpecs) > 0 {
-			return nil, fmt.Errorf("job with index %v has both pod spec and pod spec list specified", i)
+			return nil, fmt.Errorf("[createJobs] jobs must specify either a pod spec. or pod spec. list, but the %d-th job of job set %s specifies both", i, request.JobSetId)
```

Comment:
Maybe we should say jobs must specify either `podSpec` or `podSpecs`. I find it confusing to read currently

Reply:
Agreed. I changed it.